File: src/data/courses/dsa/linked-lists/singly-ll-core/creation-traversal/code/python/creation_traversal.py
# ============================================================
# Linked List Creation & Traversal
# ============================================================
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def createList(self, values):
        dummy = ListNode(0)
        tail = dummy
        for v in values:
            tail.next = ListNode(v)
            tail = tail.next
        return dummy.next

    def traverse(self, head):
        result = []
        while head:
            result.append(head.val)
            head = head.next
        return result

File: src/data/courses/dsa/linked-lists/singly-ll-core/creation-traversal/code/java/CreationTraversal.java
// ============================================================
// Linked List Creation & Traversal
// ============================================================
import java.util.*;

class ListNode { int val; ListNode next; ListNode(int v) { val = v; } }

class Solution {
    public ListNode createList(int[] values) {
        ListNode dummy = new ListNode(0), tail = dummy;
        for (int v : values) { tail.next = new ListNode(v); tail = tail.next; }
        return dummy.next;
    }

    public List<Integer> traverse(ListNode head) {
        List<Integer> result = new ArrayList<>();
        while (head != null) { result.add(head.val); head = head.next; }
        return result;
    }
}
